sig-contributor-experience/contribex-review-guidelines-on-test-infra.md

```diff
@@ -0,0 +1,123 @@
+# Guiding policy for test-infra changes requiring review by SIG Contributor Experience
+
+This document serves as a guiding policy to help identify which changes in the `git.k8s.io/test-infra/` repository require a review from the Technical Leads of SIG Contributor Experience.
+
+It also outlines areas that are outside the scope of SIG Contributor Experience's review.
+
+Note: This document is not a binding or an exhaustive list, but aims to provide general guidance.
+
+---
+
+## Index
+
+1. [When a Review from SIG Contributor Experience is Required](#when-a-review-from-sig-contributor-experience-is-required)
+2. [Recommended Areas for ContribEx Review](#recommended-areas-for-contribex-review)
+    1. [Changes to `config/plugins.yaml`](#changes-to-configpluginsyaml)
+        - [Plugins under ContribEx Review](#plugins-under-contribex-review)
+        - [Plugins Configured for Specific Orgs/Repos](#plugins-configured-for-specific-orgsrepos)
+        - [External Plugins Configured for Specific Orgs/Repos](#external-plugins-configured-for-specific-orgsrepos)
+    2. [Changes to `config/config.yaml`](#changes-to-configconfigyaml)
+    3. [Changes to `config/prow/OWNERS` File](#changes-to-configprowowners-file)
+    4. [Changes to Prow CI Jobs and Their OWNERS Files](#changes-to-prow-ci-jobs-and-their-owners-files)
+    5. [Changes to `Label_sync` Path](#changes-to-label_sync-path)
+3. [Out of Scope](#out-of-scope)
+
+---
+
+## When a Review from SIG Contributor Experience is Required
+
+A review from SIG Contributor Experience is necessary if any changes introduced by a pull request to the test-infra repo:
+
+- Introduce changes to how humans interact with Prow and its components (changes that alter contributor user experience)
+- Introduce changes to how bot interact with Prow? (TODO: verify!)
+- Introduce changes to how Prow interacts with GitHub, or what information it broadcasts back to GitHub (in form of comments, etc, for example)
+- Alter the contribution workflows for project contributors.
+
+Examples include – adding new labels like `kind/foo` enabled/required by default, or changes that modify the issue/PR triage workflow used by project maintainers.
+
+---
+
+## Recommended Areas for ContribEx Review
+
+### Changes to [`config/prow/plugins.yaml`](https://github.com/kubernetes/test-infra/blob/52fb24b56def58e3cc36b10802aa6dea230584a7/config/prow/plugins.yaml)
+
+#### Plugins under ContribEx Review
+
+Following plugins fall under ContribEx review:
```

Comment:
If we are scoping plugins under ContribEx's purview, we should probably also include the easter-eggy plugins here as well like:
- shrug
- joke
- cat
- dog
- pony
- shrug
- heart

Mainly because some of these pick images/content randomly from a pre-existing set and we want to make sure that set deosn't change, and if it does, it is an acceptable change.

Reply:
I don't think trying to enumerate the plugins makes sense.

Any plugin can introduce contributor facing behavior changes, and making changes to any of these plugins can not involve a contributor workflow change (e.g. a simple bug fix).

I would again argue that all of these plugin implementations are owned by SIG testing while contributor experience has previously asserted ownership over the contributor workflow (and similarly googlegroups automation is owned by SIG K8s Infra while contribex has asserted some ownership over some of the groups).

What we lack currently is anything written that actually says "contributor experience owns contributor workflows and you need to get approval for workflow changes" with some examples.

If you're going to unilaterally own implementations, OWNERS files and subprojects are the correct way to declare this, not another doc in k/community.